packages/xstate-store/src/types.ts

```diff
@@ -340,3 +346,18 @@ export type Selector<TContext, TSelected> = (context: TContext) => TSelected;
 export interface Selection<TSelected> extends Subscribable<TSelected> {
   get: () => TSelected;
 }
+
+export interface Readable<T> extends Subscribable<T> {
+  get(): T;
+}
+
+export interface Atom<T> extends Subscribable<T>, Readable<T> {
+  /** Gets the current value of the atom. */
+  get(): T;
+  /** Sets the value of the atom using a function. */
+  set(fn: (prevVal: T) => T): void;
+  /** Sets the value of the atom. */
+  set(value: T): void;
```

Comment:
For improved type-safety let's use non-bivariant functions instead of bivariant methods
```suggestion
  /** Sets the value of the atom using a function. */
  set: (fn: (prevVal: T) => T) => void;
  /** Sets the value of the atom. */
  set: (value: T) => void;
```

Reply:
I can't do this since overloads are only possible with `fn(...): ...` AFAIK

Reply:
You could declare it like this:
```ts
export interface Atom<T> extends Subscribable<T>, Readable<T> {
  set: {
    /** Sets the value of the atom using a function. */
    (fn: (prevVal: T) => T): void;
    /** Sets the value of the atom. */
    (value: T): void;
  };
}
```

buuuut that actually breaks things and the current types sort-of work. From types PoV, this `set` signature is somewhat weird given it both consumes and produces `T` in the first overload. So, all in all, I think we can't do any better here.